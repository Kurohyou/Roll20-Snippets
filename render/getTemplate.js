const fs = require('fs/promises');

const getTemplate = (filePath) => fs.readFile(filePath,'utf8')
.then(text => {
  return text.replace(/include k-scaffold/g,'include /node_modules/@kurohyou/k-scaffold/_k.pug')
});

module.exports = getTemplate;