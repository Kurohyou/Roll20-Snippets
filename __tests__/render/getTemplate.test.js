import { describe, it, expect, vi } from 'vitest';

import { fsSpy } from './mocks';

import getTemplate from '../../lib/render/getTemplate';

describe('getTemplate()',()=>{
  it('Should read the file and replace k-scaffold includes with full path',async ()=>{

    const spy = fsSpy.readFile(
      `include k-scaffold
+tabs({})`
    )

    const temp = await getTemplate('./mock/path');
    expect(temp).toMatchSnapshot();
  })
});