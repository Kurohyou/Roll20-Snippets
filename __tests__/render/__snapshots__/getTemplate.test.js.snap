// Vitest Snapshot v1

exports[`getTemplate() > Should read the file and replace k-scaffold includes with full path 1`] = `
"include /node_modules/@kurohyou/k-scaffold/_k.pug
+tabs({})"
`;
